packages/medusa/src/api-v2/store/products/middlewares.ts

```diff
@@ -0,0 +1,67 @@
+import { ProductStatus } from "@medusajs/utils"
+import { MiddlewareRoute } from "../../../loaders/helpers/routing/types"
+import { authenticate } from "../../../utils/authenticate-middleware"
+import { maybeApplyLinkFilter } from "../../utils/maybe-apply-link-filter"
+import {
+  applyDefaultFilters,
+  filterByValidSalesChannels,
+  setPricingContext,
+} from "../../utils/middlewares"
+import { validateAndTransformQuery } from "../../utils/validate-query"
+import * as QueryConfig from "./query-config"
+import {
+  StoreGetProductsParams,
+  StoreGetProductsParamsType,
+} from "./validators"
+
+export const storeProductRoutesMiddlewares: MiddlewareRoute[] = [
+  {
+    method: "ALL",
+    matcher: "/store/products*",
+    middlewares: [
+      authenticate("store", ["session", "bearer"], {
```

Comment:
Again, for another time - I think having a global middleware that just populates the context will be much better than relying on `authenticate`, as this middleware ideally should do actual authentication (and the `allowUnauthenticated` flag should not exist).